test/functional/spam2_controller_test.rb

```diff
@@ -228,4 +249,70 @@ def teardown
         assert_equal "Only admins and moderators can batch delete.", flash[:error]
         assert_redirected_to "/dashboard"
     end
+    #flag Nodes
+    test 'Anyone user should be able to flag a note' do
+        UserSession.create(users(:bob))
+        node = nodes(:spam)
+        post :flag_node, params: { id: node.id }
+        node = assigns(:node)
+        assert_equal 1, node.flag
+        assert_equal 'Node flagged.', flash[:notice]
+    end    
+    #unflag nodes
+    test 'Normal user should not be able to unflag a note' do
+        UserSession.create(users(:bob))
+        node = nodes(:about)
+        post :remove_flag_node, params: { id: node.id }
+        assert_equal 'Only admins and moderators can unflag nodes.', flash[:error]
+        assert_redirected_to '/dashboard'
+    end
+
+    test 'Moderator should be able to unflag nodes' do
+        UserSession.create(users(:moderator))
+        node = nodes(:about)
+        post :remove_flag_node, params: { id: node.id }
+        node = assigns(:node)
+        assert_equal 0, node.flag
+    end
+      
+    test 'Admins should be able to unflag nodes' do
+        UserSession.create(users(:admin))
+        node = nodes(:about)
+        post :remove_flag_node, params: { id: node.id }
+        node = assigns(:node)
+        assert_equal 0, node.flag
+    end
+    #flag Comments
+    test 'Anyone user should be able to flag a comments' do
```

Comment:
Same here  'Any user  should...' or 'Anyone should...'

Reply:
Done Thanks!!